### PR TITLE
fix: Profile Query Keys, User Permissions loading state, and read_only PAT creation

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
- Fixes profile query keys causing some mutations to not invalidate the correct query key
- Relates to https://github.com/linode/manager/pull/9987
- I left `useProfile` alone because the parent/child project needs `headers` functionality. Other profile query keys were reverted to their old format
- This bug is a perfect example of why we need a query key factory 🏭😖
- Also fixes:
   - An unrelated bug with User Permission loading state
   - An unrelated bug with `read_only` PAT resulting in an error: `Invalid OAuth scope: child_account:read_only`

## How to test 🧪

Query keys:
- Check out this PR
- Test profile related features such as...
  - Revoking a trusted device
  - Enabling / disabling 2FA
  - Creating, updating, and deleting SSH keys
- Verify the changes to `useGrants` does not break anything 
- Do you best to verify all query keys "match up"

Loading state:
- Have a restricted user on your account.
- Go to http://localhost:3000/account/users and select the "User Permissions" for that restricted user
- Confirm that the loading spinner displays while the queries load and then the Full Account Access is **disabled and toggled to false**; we should not see the page load with the Full Account Access toggled to true and enabled and then later change to false (the bug)

`read_only` PAT error:
- Go to http://localhost:3000/profile/tokens
- Create a Personal Access Token and select all to 'Read Only'
- Confirm that you can create a new PAT with Read Only permissions

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/queries/profile.ts [hunk 1]
```diff
@@ -60,28 +60,25 @@ export const updateProfileData = (
   newData: Partial<Profile>,
   queryClient: QueryClient
 ): void => {
-  queryClient.setQueryData([queryKey, undefined], (oldData: Profile) => ({
+  queryClient.setQueryData([queryKey, {}], (oldData: Profile) => ({
```
Comment (bnussman-akamai) [inline packages/manager/src/queries/profile.ts:63]: `useProfile` will be `[queryKey, {}]`, **not** `[queryKey, undefined]`. This query key was incorrect

File: packages/manager/src/features/Users/UserPermissions.tsx [hunk 2]
```diff
@@ -113,12 +114,13 @@ class UserPermissions extends React.Component<CombinedProps, State> {
...
 
   render() {
+    const { loading } = this.state;
     const { username } = this.props;
 
     return (
       <React.Fragment>
         <DocumentTitleSegment segment={`${username} - Permissions`} />
-        {this.renderBody()}
+        {loading ? <CircleProgress /> : this.renderBody()}
```
Comment (mjac0bs) [inline packages/manager/src/features/Users/UserPermissions.tsx:123]: This addresses a regression in the User Permissions page introduced in [here](https://github.com/linode/manager/pull/10005/files#diff-ba683f131f5e31776732082d7516053cb18f029960abb101f31027b8c575fd6eL101-L107) where eliminating the loading state check is causing the User Permissions page for a restricted user to display as Full Account Access set to true and then update to false once the data loads.

**Review by dwiley-akamai — APPROVED** [review]
Fix for profile-related features such as CRUD operations for SSH Keys ✅
Fix for "User Permissions" page loading state ✅

**Comment by abailly-akamai** [comment]
@bnussman @mjac0bs the `two-factor-auth` e2e failure seems legit?

**Comment by bnussman-akamai** [comment]
@abailly-akamai I believe I've pushed a fix. Parent child folks, please double check my work. I'm not super familiar with how `useProfile` should work anymore 😵‍💫

**Review by abailly-akamai — APPROVED** [review]
Confirm the `two-factor-auth` is now passing locally and both fixes (query and loading state) are good  as far as my testing went (authentication methods, SSH keys, trusted device)

File: packages/manager/src/features/Profile/APITokens/CreateAPITokenDrawer.tsx [hunk 3]
```diff
@@ -168,6 +168,18 @@ export const CreateAPITokenDrawer = (props: Props) => {
...
+  // because it needs to be feature-flagged. Therefore, we're manually adding it here.
+  if (flags.parentChildAccountAccess && user?.user_type !== null) {
+    const childAccountIndex = allPermissions.findIndex(
+      ([scope]) => scope === 'child_account'
+    );
+    if (childAccountIndex === -1) {
+      allPermissions.push(['child_account', 0]);
+    }
+    basePermNameMap.child_account = 'Child Account Access';
+  }
```
Comment (mjac0bs) [inline packages/manager/src/features/Profile/APITokens/CreateAPITokenDrawer.tsx:182]: This was a bug caused by the fact that we can't yet add `child_account` to the scopes until it is returned by API, so we have to add it in the drawers for now.

image

**Review by jaalah-akamai — APPROVED** [review]

**Review by jdamore-linode — APPROVED** [review]

**Review by mjac0bs — APPROVED** [review]
`useProfile` looks good. SSH key CRUD, 2FA, and trusted devices was working as expected. Didn't see any unintended issues with the other queries.

**Pull request merged**